query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -45,9 +44,32 @@ class SumOverTimeFunction extends RangeFunction {
   }
 }
 
-class CountOverTimeFunction extends RangeFunction {
+abstract class SumOverTimeChunkedFunction(var sum: Double = 0d) extends ChunkedRangeFunction {
+  override final def reset(): Unit = { sum = 0d }
+  final def apply(endTimestamp: Long, sampleToEmit: TransientRow): Unit = {
+    sampleToEmit.setValues(endTimestamp, sum)
+  }
+}
 
-  var count = 0d
+class SumOverTimeChunkedFunctionD extends SumOverTimeChunkedFunction() with ChunkedDoubleRangeFunction {
+  final def addTimeDoubleChunks(doubleVect: BinaryVector.BinaryVectorPtr,
+                                doubleReader: bv.DoubleVectorDataReader,
+                                startRowNum: Int,
+                                endRowNum: Int): Unit = {
+    sum += doubleReader.sum(doubleVect, startRowNum, endRowNum)
+  }
+}
+
+class SumOverTimeChunkedFunctionL extends SumOverTimeChunkedFunction() with ChunkedLongRangeFunction {
+  final def addTimeLongChunks(longVect: BinaryVector.BinaryVectorPtr,
+                              longReader: bv.LongVectorDataReader,
+                              startRowNum: Int,
+                              endRowNum: Int): Unit = {
+    sum += longReader.sum(longVect, startRowNum, endRowNum)
```

Comment:
Because of the absence of NA in our vectors, we use Double.NaN to identify no value in our result vectors today. Without this there would be some important holes in the query processing because we cannot identify "no value".

Since long values do not have a NaN, we need a replacement for "no-value" representation.

Reply:
See the comment above.  Note that NaN is NOT the same thing as "no value".  A client can legitimately send NaN in theory, even for Prometheus which uses a very specific NaN value (there are billions of double values that can be interpreted as NaN).    We have NA capability in our vectors actually, but it is not used.  End of time series should be treated as its own problem though and using NA is not the most optimal solution.

Reply:
My comment was mis-understood.  This has no relation to End-of-Time-Series markers. This is a totally different topic which is relevant to our implementation of query processing.

In our result RangeVectors today, NaN means no-value. For example, if there is no last sample within 5 minutes, we set the value in result vector to NaN - reason being that NA is not supported in our result vectors.

My question is how are we going to set NaN on a Long based result vector ?

Reply:
Oh, I see.  This is handled correctly for LastSampleChunkedFunction even for Long's.  Basically the value returned is always NaN unless there is an actual value found.

However we should think about how to properly handle these for other types like Histograms.  I think using NaN is probably not a viable long term solution, but that solution is also out of scope for this PR.